lib/modules/hosts/youtube.js

```diff
@@ -5,7 +5,7 @@ export default {
 	name: 'youtube',
 	attribution: false,
 	domains: ['youtube.com', 'youtu.be'],
-	detect: ({ href }) => (/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href),
+	detect: ({ href }) => (/^https?:\/\/(?:www\.|m\.)?(?:youtu\.be\/|youtube\.com\/?(?:watch|embed)(?:.*?v=|v\/|\/))([\w-]+)/i).exec(href),
```

Comment:
Actually, there is one thing: I believe we need to replace `watch|embed` with `watch|embed|v` and remove the `v\/` from `(?:.*?v=|v\/|\/)`, so that links like https://youtube.com/v/iwGFalTRHDA will work.